R/jaspHistogram.R

```diff
@@ -22,25 +22,34 @@
 #' @param densityShade, logical, should the area underneath the density be shaded?
 #' @param densityShadeAlpha, numeric in \[0, 1\], transparancy for the shaded density.
 #' @param densityLineWidth, positive number, the line width of the superimposed density.
+#' @param hideXAxisLabels, logical, should the x-axis line be hidden? Defaults to \code{FALSE}.
 #' @param hideYAxisLabels, logical, should the y-axis line be hidden? Defaults to \code{showDensity}.
+#' @param hideXAxisName, logical, should the x-axis name be hidden? Defaults to \code{FALSE}.
+#' @param hideYAxisName, logical, should the y-axis name be hidden? Defaults to \code{FALSE}.
+#' @param xBreaks custom x-axis breaks.
 #' @example inst/examples/ex-jaspHistogram.R
 #' @export
 jaspHistogram <- function(
   x, xName,
-  groupingVariable = NULL,
+  groupingVariable  = NULL,
   groupingVariableName,
-  histogram = TRUE,
+  histogram         = TRUE,
   histogramPosition = "dodge",
-  binWidthType = c("doane", "fd", "scott", "sturges", "manual"),
-  numberOfBins = NA,
-  rugs = FALSE,
-  rugsColor = FALSE,
-  density = FALSE,
-  densityColor = FALSE,
-  densityShade = FALSE,
+  binWidthType      = c("doane", "fd", "scott", "sturges", "manual"),
+  numberOfBins      = NULL,
+  rugs              = FALSE,
+  rugsColor         = FALSE,
+  density           = FALSE,
+  densityColor      = FALSE,
+  densityShade      = FALSE,
   densityShadeAlpha = 0.6,
-  densityLineWidth = 1,
-  hideYAxisLabels = density) {
+  densityLineWidth  = 1,
+  hideXAxisLabels   = FALSE,
+  hideYAxisLabels   = density,
+  hideXAxisName     = FALSE,
+  hideYAxisName     = FALSE,
+  xBreaks           = NULL
```

Comment:
do you mean only `xBreaks` or the other arguments in general?

the `xBreaks` were added so that we can pass the same breaks into different plotting functions in the matrix or bivariate with margins plots. I do not remember about the rest of the args but I can try to read the code a bit more and remember.

but, iirc, I wasn't happy with the jaspHistogram changes I did there to make it work with the jaspMatrixPlot, that's why there's also the new jaspMarginal functionality. Specifically, I think ideally we would use only the jaspMarginal, jaspBivariate, jaspBivariateWithMargins, and jaspMatrixPlot together and revert the changes to jaspHistogram, so that analyses that already used jaspHistogram will still work normally. But I think the jaspMarginal is not fully working so there's still some work to be done there 🥲